waveorder/io/reader.py

```diff
@@ -0,0 +1,158 @@
+import numpy as np
+import os
+import zarr
+from tifffile import TiffFile
+
+
+class MicromanagerReader:
+
+    def __init__(self, folder):
+        """
+        loads ome-tiff files in folder into zarr or numpy arrays
+        Strategy:
+            1. read all files in folder
+            2. search for the file whose omexml does not contain the element tag "BinaryOnly"-> this is the master ome-tiff
+            3. each series represents a micro-manager stage position.  upon call for data, assign this data to a dict
+
+        :param folder: folder containing all ome-tiff files
+        :param reader: tifffile or aicsimageio
+        """
+
+        self._positions = {}
+        self.mm_meta = None
+        self.stage_positions = 0
+        self.height = 0
+        self.width = 0
+        self.frames = 0
+        self.slices = 0
+        self.channels = 0
+        self.chnames = []
+
+        self.master_ome_tiff = self._get_master_ome_tiff(folder)
+        print(self.master_ome_tiff)
+        self._set_mm_meta()
+
+    def _set_mm_meta(self):
```

Comment:
Micromanager writes Summary metadata and per-image metadata. It would be useful to be able to retrieve metadata for a specific image at given TCZYX coordinates. I tried doing something like that with this function, it may need some debugging.

https://github.com/czbiohub/aicsimageio/blob/30fa17a1fa0b1e9351dea610972e438c87185a4a/aicsimageio/readers/micromanager_reader.py#L114-L132

Reply:
This could be a cool feature, but perhaps it deserves another PR.  Your link references the current state of aicsimageio, which uses [its own OME class](https://github.com/czbiohub/aicsimageio/blob/30fa17a1fa0b1e9351dea610972e438c87185a4a/aicsimageio/vendor/omexml.py#L302) to parse the files.  The future of aicsimageio uses [`ome_types` library ](https://github.com/tlambert03/ome-types).

The worry is that, ome_types threw a bunch of errors when loading micro-manager data when using aicsimageio.  If we use it here, it might also give us problems.  Will need to dive deeper.

Reply:
I see, I haven't kept up to date with the way metadata reading has changed. We can certainly leave then for a future PR